SimPEG/electromagnetics/static/utils/static_utils.py

```diff
@@ -552,132 +560,129 @@ def plot_3d_pseudosection(
-            plane_points = [plane_points]
-
-        # Expand to list of only one plane distance for all planes
-        if isinstance(plane_distance, list) != True:
-            plane_distance = len(plane_points) * [plane_distance]
 
-        # Pre-allocate index for points on plane(s)
-        k = np.zeros(len(dvec), dtype=bool)
-        for ii in range(0, len(plane_points)):
-
-            p1, p2, p3 = plane_points[ii]
-            a, b, c, d = define_plane_from_points(p1, p2, p3)
-
-            k = k | (
-                np.abs(a * midxy[:, 0] + b * midxy[:, 1] + c * midz + d)
-                / np.sqrt(a ** 2 + b ** 2 + c ** 2)
-                < plane_distance[ii]
-            )
-
-        if np.all(k == 0):
-            raise Exception(
-                """No locations are within *plane_distance* of any plane(s)
-                defined by *plane_points*. Try increasing *plane_distance*."""
-            )
-
-        if vlim == None:
-            norm = mpl.colors.Normalize(vmin=dvec[k].min(), vmax=dvec[k].max())
-        else:
-            norm = mpl.colors.Normalize(vmin=vlim[0], vmax=vlim[1])
+    # Filled contour plot
+    elif plot_type == 'tricontourf':
 
-        data_plot = ax.scatter(
-            midxy[k, 0],
-            midxy[k, 1],
-            midz[k],
-            dvec[k],
-            s=s,
-            c=dvec[k],
-            depthshade=False,
+        # Make initial contour plot
+        data_plot = ax.tricontourf(
+            locations[:, 0],
+            locations[:, -1],
+            dvec,
             norm=norm,
-            **scatter_opts,
+            **tricontourf_opts,
         )
 
+    else:
+        raise NotImplementedError("plot_type must be 'scatter' or 'tricontourf'")
+        
+    # Use a filled polygon to mask everything above
+    # that has a pseudo-location above the positions
+    # for nearest electrode spacings    
+    
+    if mask_topography:
+        
+        electrode_locations = np.unique(np.r_[
+            survey.locations_a,
+            survey.locations_b,
+            survey.locations_m,
+            survey.locations_n
+        ], axis=0)
+        
+        zmin = np.min(electrode_locations[:, 1])
+        zmax = np.max(electrode_locations[:, 1])
+
+        tree = cKDTree(locations)
+        _, nodeInds = tree.query(electrode_locations)
+
+        poly_locations = locations[nodeInds, :]
+
+        poly_locations = np.r_[
+            np.c_[np.min(poly_locations[:, 0]), zmax],
+            poly_locations,
+            np.c_[np.max(poly_locations[:, 0]), zmax]
+        ]
+
+        ax.fill(
+            poly_locations[:, 0], poly_locations[:, 1],
+            facecolor='w', linewidth=0.5
+        )    
+    
+    z_top = np.max(locations[:, -1])
+    z_bot = np.min(locations[:, -1])
+    ax.set_ylim(z_bot - 0.03*(z_top-z_bot), z_top + 0.03*(z_top-z_bot))
+    ax.set_xlabel("Line position (m)")
+    ax.set_ylabel("Pseudo-elevation (m)")
+
     # Define colorbar
     if create_colorbar:
```

Comment:
I want to see us use a LogNorm for log scales instead of doing this odd thing of plotting the log values of data and hacking the colorbar formats. I would like to see re-use of the 2D plotting utility in SimPEG/utilities/plot_utils.py instead of yet another recreating of 2D plotting.

Reply:
I tried several ways to implement this but there is an issue in mpl for customizing the ticks on the colorbar when doing a log plot. The original ticks on the colorbar are hard coded somehow when setting the mapping to LogNorm. The new ticks just plot on top of them as you can see below. I literally tried to do cbar.set_ticks([]) to get rid of any ticks. For linear scale, they would disappear. For log-norm they do not. Until this bug is fixed, I am inclined to leave it the way it is because it is stable.

If the data always span several orders of magnitude there isn't much point in playing with this. But if you want to plot app. res. and the values only span 1 order of magnitude, it isn't good.

![image](https://user-images.githubusercontent.com/12970009/113516504-742c6480-952f-11eb-85a9-976ee9d13c8e.png)